Mathlib/Topology/Category/Profinite/Limits.lean

```diff
@@ -0,0 +1,211 @@
+  simp only [Limits.limit.conePointUniqueUpToIso_hom_comp, pullback.cone_pt, pullback.cone_π]
+
+theorem explicit_snd_eq :
+    Profinite.pullback.snd f g = (pullbackIsoPullback f g).hom ≫ Limits.pullback.snd := by
+  dsimp [pullbackIsoPullback]
+  simp only [Limits.limit.conePointUniqueUpToIso_hom_comp, pullback.cone_pt, pullback.cone_π]
+
+end Isos
+
+end Pullbacks
+
+section FiniteCoproducts
+
+variable {α : Type} [Fintype α] (X : α → Profinite.{u})
+
+/--
+The coproduct of a finite family of objects in `Profinite`, constructed as the disjoint
+union with its usual topology.
+-/
+def finiteCoproduct : Profinite := Profinite.of <| Σ (a : α), X a
+
+/-- The inclusion of one of the factors into the explicit finite coproduct. -/
+def finiteCoproduct.ι (a : α) : X a ⟶ finiteCoproduct X where
+  toFun := (⟨a, ·⟩)
+  continuous_toFun := continuous_sigmaMk (σ := fun a => X a)
+
+/--
+To construct a morphism from the explicit finite coproduct, it suffices to
+specify a morphism from each of its factors.
+This is essentially the universal property of the coproduct.
+-/
+def finiteCoproduct.desc {B : Profinite.{u}} (e : (a : α) → (X a ⟶ B)) :
+    finiteCoproduct X ⟶ B where
+  toFun := fun ⟨a, x⟩ => e a x
+  continuous_toFun := by
+    apply continuous_sigma
+    intro a
+    exact (e a).continuous
+
+@[reassoc (attr := simp)]
+lemma finiteCoproduct.ι_desc {B : Profinite.{u}} (e : (a : α) → (X a ⟶ B)) (a : α) :
+  finiteCoproduct.ι X a ≫ finiteCoproduct.desc X e = e a := rfl
+
+lemma finiteCoproduct.hom_ext {B : Profinite.{u}} (f g : finiteCoproduct X ⟶ B)
+    (h : ∀ a : α, finiteCoproduct.ι X a ≫ f = finiteCoproduct.ι X a ≫ g) : f = g := by
+  ext ⟨a, x⟩
+  specialize h a
+  apply_fun (· x) at h
+  exact h
+
+/-- The coproduct cocone associated to the explicit finite coproduct. -/
+@[simps]
+def finiteCoproduct.cocone : Limits.Cocone (Discrete.functor X) where
+  pt := finiteCoproduct X
+  ι := Discrete.natTrans fun ⟨a⟩ => finiteCoproduct.ι X a
+
+/-- The explicit finite coproduct cocone is a colimit cocone. -/
+@[simps]
+def finiteCoproduct.isColimit : Limits.IsColimit (finiteCoproduct.cocone X) where
+  desc := fun s => finiteCoproduct.desc _ fun a => s.ι.app ⟨a⟩
+  fac := fun s ⟨a⟩ => finiteCoproduct.ι_desc _ _ _
+  uniq := fun s m hm => finiteCoproduct.hom_ext _ _ _ fun a => by
+    specialize hm ⟨a⟩
+    ext t
+    apply_fun (· t) at hm
+    exact hm
+
+section Iso
+
+noncomputable
+def coproductIsoCoproduct : finiteCoproduct X ≅ ∐ X :=
+Limits.IsColimit.coconePointUniqueUpToIso (finiteCoproduct.isColimit X) (Limits.colimit.isColimit _)
+
+theorem Sigma.ι_comp_toFiniteCoproduct (a : α) :
+    finiteCoproduct.ι X a = (Limits.Sigma.ι X a) ≫ (coproductIsoCoproduct X).inv := by
+  simp only [coproductIsoCoproduct, Limits.colimit.comp_coconePointUniqueUpToIso_inv,
+    finiteCoproduct.cocone_pt, finiteCoproduct.cocone_ι, Discrete.natTrans_app]
+
+noncomputable
+def coproductHomeoCoproduct : finiteCoproduct X ≃ₜ (∐ X : _) :=
```

Comment:
It feels inconsistent to have `coproductHomeoCoproduct` but not `pullbackHomeoPullback`. Unless you have some immediate use for it, I would just remove `coproductHomeoCoproduct` for now, considering that you don't develop any API for it. If it turns out to be useful later, it can still be added again together with the necessary API.

Reply:
It will be used in upcoming PR's developing condensed mathematics. But I can add `pullbackHomeoPullback` for consistency.